extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -117,6 +117,115 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
+        //     "cache-control": "",
+        //     "content-disposition": "",
+        //     "content-encoding": "",
+        //     "expires": "",
+        //     "content-length": 231,
+        //     "content-type": "",
+        //     "content-md5": "13bc3e1ff93133bc45aff53912404f74",
+        //     "content-language": "",
+        //     "creation-time": "2022-06-03T14:13:17.852Z",
+        //     "x-amz-version-id": "null",
+        //     "x-amz-server-version-id": "",
+        //     "x-amz-storage-class": "aws-location",
+        //     "x-amz-server-side-encryption": "",
+        //     "x-amz-server-side-encryption-aws-kms-key-id": "",
+        //     "x-amz-server-side-encryption-customer-algorithm": "",
+        //     "x-amz-website-redirect-location": "",
+        //     "acl": {
+        //         "Canned": "private",
+        //         "FULL_CONTROL": [],
+        //         "WRITE_ACP": [],
+        //         "READ": [],
+        //         "READ_ACP": []
+        //     },
+        //     "key": "key11",
+        //     "location": [
+        //         {
+        //         "key": "bucket1/key11",
+        //         "size": 231,
+        //         "start": 0,
+        //         "dataStoreName": "aws-location",
+        //         "dataStoreType": "aws_s3",
+        //         "dataStoreETag": "1:13bc3e1ff93133bc45aff53912404f74",
+        //         "dataStoreVersionId": "WrMzKu4YCYv6hDeLNtSDLoAlGRGaxpCT"
+        //         }
+        //     ],
+        //     "isDeleteMarker": false,
+        //     "tags": {},
+        //     "replicationInfo": {
+        //         "status": "",
+        //         "backends": [],
+        //         "content": [],
+        //         "destination": "",
+        //         "storageClass": "",
+        //         "role": "",
+        //         "storageType": "",
+        //         "dataStoreVersionId": "",
+        //         "isNFS": null
+        //     },
+        //     "dataStoreName": "aws-location",
+        //     "originOp": "s3:ObjectRestore",
+        //     "last-modified": "2022-06-03T14:13:17.852Z",
+        //     "md-model-version": 5,
+        //     "archive": {
+        //         "restoreRequestedAt": "2022-06-03T17:20:38.183Z",
+        //         "restoreRequestedDays": 10
+        //     }
+        // }
+
+        // TODO: We will need to add `objectMD.originOp = 's3:ObjectRestore';`
+        // in CloudServer object restore API
+        const operation = value.originOp;
+        if (operation !== 's3:ObjectRestore') {
+            return;
+        }
+
+        const locationName = value.dataStoreName;
+
+        // if object already restore nothing to do, S3 has already updated object MD
+        const isObjectAlreadyRestored = value.archive
+        && value.archive.restoreCompletedAt
+        && new Date(value.archive.restoreWillExpireAt) >= new Date(Date.now());
```

Comment:
```suggestion
        && new Date(value.archive.restoreWillExpireAt) >= new Date();
```
new date returns the now date

Reply:
This PR is supposed to be for developers to know what is needed and where things are supposed to land.
I have no intention to merge it :)